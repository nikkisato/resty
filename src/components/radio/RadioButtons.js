import React from 'react';
import PropTypes from 'prop-types';
import RadioButton from './RadioButton';

const RadioButtons = ({  method, onChange }) => {
  const radioButtons = [
    { label: 'Put', value: 'PUT' },
    { label: 'Get', value: 'GET' },
    { label: 'Patch', value: 'PATCH' },
    { label: 'Delete', value: 'DELETE' },
    { label: 'Post', value: 'POST' }
  ];
  const radioButtonElements = radioButtons.map(({ label, value }) => (
    <RadioButton
      key={value}
      label={label}
      value={value}
      method={method}
      onChange={onChange}
    />
  ));
  return <>{radioButtonElements}</>;
};

RadioButton.propTypes = {
  radioButtons: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired
    })
  ).isRequired,
  onChange: PropTypes.func.isRequired
};

export default RadioButtons;
