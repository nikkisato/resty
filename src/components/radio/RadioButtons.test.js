
import React from 'react';
import { shallow } from 'enzyme';
import RadioButtons from './RadioButtons';

describe('RadioButtons component', () => {
  it('renders RadioButtons', () => {
    const wrapper = shallow(<RadioButtons />);
    expect(wrapper).toMatchSnapshot();
  });
});
