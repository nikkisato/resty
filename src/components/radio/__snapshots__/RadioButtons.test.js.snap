// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtons component renders RadioButtons 1`] = `
<Fragment>
  <RadioButton
    key="GET"
    label="Get"
    value="GET"
  />
  <RadioButton
    key="POST"
    label="Post"
    value="POST"
  />
  <RadioButton
    key="PUT"
    label="Put"
    value="PUT"
  />
  <RadioButton
    key="PATCH"
    label="Patch"
    value="PATCH"
  />
  <RadioButton
    key="DELETE"
    label="Delete"
    value="DELETE"
  />
</Fragment>
`;
